fixed styles for job-rb

dashu-baba opened the pull request.

maxceem commented on src/components/FormField/index.jsx:
```diff
@@ -36,8 +36,14 @@ const FormField = ({ field, isGroupField }) => {
-              className={meta.error ? "error" : ""}
+              className={meta.error && meta.touched ? "error" : ""}
               readonly={field.readonly}
+              onBlur={(event) => {
+                input.onBlur(event);
+              }}
+              onFocus={(event) => {
+                input.onFocus(event);
+              }}
```
Can we directly pass like `onBlur={input.onBlur}` and `onFocus={input.onFocus}`?

maxceem commented on src/components/FormField/index.jsx:
```diff
@@ -47,15 +53,27 @@ const FormField = ({ field, isGroupField }) => {
               minValue={field.minValue}
               onChange={input.onChange}
-              className={meta.error ? "error" : ""}
+              onBlur={(event) => {
+                input.onBlur(event);
+              }}
+              onFocus={(event) => {
+                input.onFocus(event);
+              }}
```
Can we directly pass like `onBlur={input.onBlur}` and `onFocus={input.onFocus}`?

maxceem commented on src/components/FormField/index.jsx:
```diff
@@ -47,15 +53,27 @@ const FormField = ({ field, isGroupField }) => {
               value={input?.value ?? ""}
               onChange={input.onChange}
-              className={meta.error ? "error" : ""}
+              onBlur={(event) => {
+                input.onBlur(event);
+              }}
+              onFocus={(event) => {
+                input.onFocus(event);
+              }}
```
Can we directly pass like `onBlur={input.onBlur}` and `onFocus={input.onFocus}`?

maxceem commented on src/components/FormField/index.jsx:
```diff
@@ -65,6 +83,12 @@ const FormField = ({ field, isGroupField }) => {
               onChange={(date) => {
                 input.onChange(date);
               }}
+              onBlur={(event) => {
+                input.onBlur(event);
+              }}
+              onFocus={(event) => {
+                input.onFocus(event);
+              }}
```
Can we directly pass like `onBlur={input.onBlur}` and `onFocus={input.onFocus}`?

maxceem commented on src/components/FormField/index.jsx:
```diff
@@ -75,9 +99,15 @@ const FormField = ({ field, isGroupField }) => {
               }}
               options={field.selectOptions}
               isMulti={field.isMulti}
+              onBlur={(event) => {
+                input.onBlur(event);
+              }}
+              onFocus={(event) => {
+                input.onFocus(event);
+              }}
```
Can we directly pass like `onBlur={input.onBlur}` and `onFocus={input.onFocus}`?

maxceem commented on src/components/ReactSelect/styles.module.scss:
```diff
@@ -10,3 +10,11 @@
+  min-height: 32px;
+}
+
+input{
+  transition: none !important;
+}
```
This style is global and would be applied to all inputs in the application. Could we make it more specific so it is only applied to the input inside Select?

maxceem commented on src/components/TCForm/styles.module.scss:
```diff
@@ -1,6 +1,13 @@
 
+
+@media (max-width: 1421px) {
+  .tc-form  > .job-form-fields {
+      padding: 0px !important;
+    }
```
Can we avoid using `important`?

maxceem commented on src/routes/JobForm/index.jsx:
```diff
@@ -78,7 +78,7 @@ const JobForm = ({ teamId, jobId }) => {
   useEffect(() => {
     if (skills && job && !options) {
-      const skillOptions = skills.map((item) => {
+      const skillOptions = skills.slice(0, 10).map((item) => {
```
I guess this is because now we see only 10 skills, we should see all the skills.

maxceem reviewed (changes requested):
@dashu-baba most issues are fixed, but some have issues, and some break something else after fixing. Also, there are places in code that I'd like to improve, please, see comments for the code.

1. Placeholder font is still different. Also, there should be no uppercase.

    ![image](https://user-images.githubusercontent.com/146016/107035404-cddb1180-67c0-11eb-94ab-ce255ed53bf0.png)

2. After this, the selected option in select is vertically aligned, but the arrow and cross now jump up. It should stay vertically aligned:

    ![image](https://user-images.githubusercontent.com/146016/107036281-0cbd9700-67c2-11eb-8607-3bdfea98b3e8.png)

3. Before the fix, skills select shows all the skills. But after the select is shows only limited number of skills with no scrollbar:

    ![image](https://user-images.githubusercontent.com/146016/107036380-3bd40880-67c2-11eb-855c-a327f2d7ffd5.png)

4. During resource booking editing there is some white gap now:

     ![image](https://user-images.githubusercontent.com/146016/107036613-953c3780-67c2-11eb-82ec-917096459da0.png)

5. Also, there are not aligned:
     
   ![image](https://user-images.githubusercontent.com/146016/107036730-bc930480-67c2-11eb-88d9-7ab633dbb1a3.png)

6. During editing Resource Booking for Client Rate and Member Rate labels are still not shown:

    ![image](https://user-images.githubusercontent.com/146016/107036823-e2b8a480-67c2-11eb-9c44-33edbaad58f0.png)
    
7. When I make the browser window smaller the form fields are jumping, see video https://monosnap.com/file/rYabiIf06iMVnOhx31DLIsmnrobRZh.
    They should not jump, they only have to be max-width 640px. And as soon as window is smaller they just shrink together with the window.

dashu-baba commented:
1. Fixed
![image](https://user-images.githubusercontent.com/44001726/107066941-cd666900-6808-11eb-9a32-5274029918c3.png)

2. Fixed.
![image](https://user-images.githubusercontent.com/44001726/107066698-80829280-6808-11eb-93f1-3b4e798fc2bd.png)

3. Fixed 
![image](https://user-images.githubusercontent.com/44001726/107067022-e707b080-6808-11eb-8d39-5aa1258ef139.png)

4. Fixed
![image](https://user-images.githubusercontent.com/44001726/107067238-3352f080-6809-11eb-8023-a67f4fbe032f.png)

5. Fixed
![image](https://user-images.githubusercontent.com/44001726/107067354-5e3d4480-6809-11eb-89f3-b6593ad7f45a.png)

6. Fixed
![image](https://user-images.githubusercontent.com/44001726/107067407-701ee780-6809-11eb-897f-872e3124e771.png)

7. Not sure if I understand correctly, I made the form max size to 640px. So it will be like that until the container with reduce. It will then reduce the padding with smooth transition.

maxceem reviewed (changes requested):
1. Spacing around skils is still changing when new row of skills is added, see demo video https://monosnap.com/file/g3nsn77R7fwvIwMcVjwHMfL4CRCMGu
    - when we have only 1 line of skills spacing is perfect
    - when I add the 2nd line of skills, the spacing above the first line become small, but it should stay the same
    - when I add the 3rd line of skills, the spacing between 1st and 2nd line changes, and it's different from the spacing between 2nd and 3rd line. But all of them should stay the same during adding and removing lines

2. Placeholder inside Select is different

    ![image](https://user-images.githubusercontent.com/146016/107115924-a0519f00-6878-11eb-9fa8-3c2dc9f31891.png)

3. **[extra +$10]** Select height is `42px`, but it should be `40px` as all other fields.
    Also, the border color of the Select is different from the provided design and other fields
    
    ![image](https://user-images.githubusercontent.com/146016/107116078-b318a380-6879-11eb-84dc-c9dac119a0ca.png)

4. > Not sure if I understand correctly, I made the form max size to 640px. So it will be like that until the container with reduce. It will then reduce the padding with smooth transition.
    
    Expected behavior is like on this video https://monosnap.com/file/pw2BhvvCdFEVfKVPaeufM5HMyGYwzR or check the live page https://connect.topcoder-dev.com/projects/16957/scope

    - Form should have `width: 100%` to make it fluid, but have `max-width: 640px` so it doesn't become too wide. And it should be centered. So we don't need any media conditions for different screen sizes. Only make width of the form fluid with max-width 640px and centered.

Also, there are few comments for the code below.

maxceem commented on src/components/TCForm/styles.module.scss:
```diff
@@ -50,3 +56,43 @@
   z-index: 100;
 }
+
+@media (max-width: 1366px) {
+  .tc-form{
+    .job-form-fields-container {
+        padding-left: 15%;
+        padding-right: 15%;
+        transition: 300ms;
+      }
+  }
+}
+
+@media (max-width: 1280px) {
+  .tc-form{
+    .job-form-fields-container {
+        padding-left: 10%;
+        padding-right: 10%;
+        transition: 300ms;
+      }
+  }
+}
+
+@media (max-width: 1020px) {
+  .tc-form{
+    .job-form-fields-container {
+        padding-left: 5%;
+        padding-right: 5%;
+        transition: 300ms;
+      }
+  }
+}
+
+@media (max-width: 854px) {
+  .tc-form{
+    .job-form-fields-container {
+        padding-left: 0;
+        padding-right: 0;
+        transition: 300ms;
```
we don't need media conditions for this functionality, everything we need is:
- make all the fuilds fluid so they take 100% width of the form
- make the form 100% width but with max-width 640px

maxceem reviewed (approved):
Thank you for the quick turn around in fixing all the small issues, @dashu-baba. All works good now.
